infoblox/resource_infoblox_ip_allocation.go

```diff
@@ -106,34 +114,42 @@ func resourceIPAllocation() *schema.Resource {
 	}
 }
 
-func getAndRenewHostRecAltId(
-	d *schema.ResourceData, m interface{}) (
-
-	hostRec *ibclient.HostRecord, err error) {
-
-	internalId, ref, err := getAltIdFields(d.Id())
-	if err != nil {
-		return nil, err
-	}
-
+func getAndRenewHostRecAltId(d *schema.ResourceData, m interface{}) (hostRec *ibclient.HostRecord, err error) {
+	internalId, ref, _ := getAltIdFields(d.Id())
 	objMgr := ibclient.NewObjectManager(m.(ibclient.IBConnector), "Terraform", "")
-	hostRec, err = objMgr.SearchHostRecordByAltId(internalId, ref, eaNameForInternalId)
+	hostRec, err = objMgr.SearchHostRecordByAltId(internalId.String(), ref, eaNameForInternalId)
 	if err != nil {
 		if _, ok := err.(*ibclient.NotFoundError); !ok {
 			return nil, fmt.Errorf(
 				"error getting the allocated host record with ID '%s': %s",
 				d.Id(), err.Error())
 		}
 		log.Printf("resource with the ID '%s' has been lost, removing it", d.Id())
+
+		// TODO: implement logging in case of an error returned, for all similar places as well
 		d.SetId("")
 		return nil, nil
 	}
-	if ref != hostRec.Ref {
-		newAltId, err := renewAltId(d.Id(), hostRec.Ref)
-		if err != nil {
-			return nil, nil
+
+	if internalId == nil {
+		if hostRec.Ea != nil {
+			rawValue, found := hostRec.Ea[eaNameForInternalId]
+			if found {
+				stringVal, ok := rawValue.(string)
+				// TODO: do we need this after changing internalId's implementation?
+				if ok && isValidInternalId(stringVal) {
```

Comment:
As of now, this is just for one place in the code, but there is no guarantee that this will not be changed in the future for some reason. That is why I decided to abstract from particular format here. Let it be defined in the code which determines alternative/internal ID's logic.